extensions/wikia/WikiFactory/tests/WikiFactoryTest.php

```diff
@@ -114,43 +122,96 @@ public function testGetLocalEnvURLDataProvider() {
 		return [
 			[
 				'env' => WIKIA_ENV_PREVIEW,
+				'forcedEnv' => null,
 				'url' => 'http://muppet.wikia.com',
 				'expected' => 'http://preview.muppet.wikia.com'
 			],
 			[
 				'env' => WIKIA_ENV_VERIFY,
+				'forcedEnv' => null,
 				'url' => 'http://muppet.wikia.com/wiki/Muppet',
 				'expected' => 'http://verify.muppet.wikia.com/wiki/Muppet'
 			],
 			[
 				'env' => WIKIA_ENV_STABLE,
+				'forcedEnv' => null,
 				'url' => 'http://muppet.wikia.com/wiki/Muppet',
 				'expected' => 'http://stable.muppet.wikia.com/wiki/Muppet'
 			],
 			[
 				'env' => WIKIA_ENV_DEV,
+				'forcedEnv' => null,
 				'url' => 'http://muppet.wikia.com/wiki',
 				'expected' => 'http://muppet.' . self::MOCK_DEV_NAME . '.wikia-dev.com/wiki'
 			],
 			[
 				'env' => WIKIA_ENV_SANDBOX,
+				'forcedEnv' => null,
 				'url' => 'http://gta.wikia.com/Vehicles_in_GTA_III',
 				'expected' => 'http://sandbox-s1.gta.wikia.com/Vehicles_in_GTA_III'
 			],
 			[
 				'env' => WIKIA_ENV_VERIFY,
+				'forcedEnv' => null,
 				'url' => 'http://gta.wikia.com/wiki/test/test/test',
 				'expected' => 'http://verify.gta.wikia.com/wiki/test/test/test'
 			],
 			[
 				'env' => WIKIA_ENV_STAGING,
+				'forcedEnv' => null,
 				'url' => 'http://gta.wikia.com/wiki/test/test/test',
 				'expected' => 'http://gta.wikia-staging.com/wiki/test/test/test'
 			],
 			[
 				'env' => WIKIA_ENV_DEV,
+				'forcedEnv' => null,
 				'url' => 'http://gta.wikia.com/',
 				'expected' => 'http://gta.' . self::MOCK_DEV_NAME . '.wikia-dev.com'
+			],
+			[
+				'env' => WIKIA_ENV_DEV,
+				'forcedEnv' => WIKIA_ENV_PREVIEW,
+				'url' => 'http://gta.wikia.com/',
+				'expected' => 'http://preview.gta.wikia.com'
+			],
+			[
+				'env' => WIKIA_ENV_PREVIEW,
+				'forcedEnv' => WIKIA_ENV_DEV,
+				'url' => 'http://gta.wikia.com/',
+				'expected' => 'http://preview.gta.wikia.com'
+			]
+		];
+	}
+
+	public function testPrepareUrlToParseDataProvider() {
+		return [
+			[
+				'http://www.community-name.wikia.com',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "http://www.community-name.wikia.com" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)